### ISSUE TITLE
Add support for logging HttpRequest and HttpResponse objects

### ISSUE THREAD

**Issue by jongpie** [jongpie/NebulaLogger#240]
### Discussed in https://github.com/jongpie/NebulaLogger/discussions/239

<div type='discussions-op-text'>

<sup>Originally posted by **guerrieroi** October 26, 2021</sup>
I would like to know if there is a best practice to log status code and message, request and response body of Apex REST callout with Nebula Logger.
I'm using logger.error(message) with a custom formatted message containing all above data about REST callout.

Thanks in advance.</div>

Logger should provide the ability to log [`HttpRequest`](https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_classes_restful_http_httprequest.htm) and [`HttpResponse`](https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_classes_restful_http_httpresponse.htm#apex_classes_restful_http_httpresponse) objects. At a minimum, these should be added as new methods in `LogEntryEventBuilder` - each method should then map attributes like the response/request's status code (and other attributes) to new fields on `LogEntryEvent__e` and `LogEntry__c`.

**Comment by jongpie** [comment]
Thanks @PawelWozniak for sharing your code snippet! I'm hoping to finally work on this issue next month, I'm hoping it will be fairly easy to implement this.

**Comment by jongpie** [comment]
I'm marking this as closed - the ability to log `HttpRequest` and `HttpResponse` objects was added in [v4.7.1](https://github.com/jongpie/NebulaLogger/releases/tag/v4.7.1) (PR #298) to the class `LogEntryEventBuilder`, but I kept this issue open because I had planned to also add overloads to `Logger` that accepted `HttpRequest` and `HttpResponse` parameters. However, at least for now, I don't think that's necessary, and it would really increase the number of loads needed in `Logger`, so I'm going to consider the builder methods to be good enough for now. If/when there is a need to add the overloads to `Logger`, a new issue can be created to track it.
